File: src/main/java/me/devksh930/lec15/Lec15Main.java
package me.devksh930.lec15;

import java.util.*;

public class Lec15Main {
    public static void main(String[] args) {
        int[] array = {100, 200};

        for (int i = 0; i < array.length; i++) {
            System.out.printf("%s %s", i, array[i]);
        }

        final List<Integer> numbers = Arrays.asList(100, 200);

        //jdk8
        HashMap<Integer, String> map = new HashMap<>();
        map.put(1, "MONDAY");
        map.put(2, "TUESDAY");
        //jdk9 이상
        Map.of(1, "MONDAY", 2, "TUESDAY");

        //키를 통해서 접근
        for (int key : map.keySet()) {
            System.out.println(key);
            System.out.println(map.get(key));
        }
        //entry를(key,value)를 가지는 객체를 이용함
        for (Map.Entry<Integer, String> entry : map.entrySet()) {
            System.out.println(entry.getKey());
            System.out.println(entry.getValue());
        }
    }
}

File: src/main/java/me/devksh930/lec15/Lec15Main.kt
package me.devksh930.lec15

import java.util.*

fun main() {
    val array = arrayOf(100, 200)
    for (i in array.indices) {
        println("${i} ${array[i]}")
    }
    array.plus(300)
    for ((idx, value) in array.withIndex()) {
        println("${idx} ${value}")
    }

    val numbers = listOf(100, 200)
    val emptyList = emptyList<Int>()

    println(numbers[0])

    for (number in numbers) {
        println(numbers)
    }


    val mutableListOf = mutableListOf(100, 200)
//    printNumber(emptyList())


    val numbersSet = setOf(100, 200)

    val oldMap = mutableMapOf<Int, String>()
    oldMap[1] = "MONDAY"
    oldMap[2] = "TUESDAY"

    mapOf(1 to "MONDAY", 2 to "TUESDAY")

    for (key in oldMap.keys) {
        println(key)
        println(oldMap[key])
    }

    for ((key, value) in oldMap.entries) {
        println(key)
        println(value)
    }
}

//private fun printNumber(number: List<Int>) {
//    TODO()
//}
//
